spec/requests/regions_spec.rb

```diff
@@ -8,40 +8,86 @@
 # GET /api/regions/:id
 #
 
-describe "Regions API" do
-  it "forbids access to regions without an appropriate role" do
-    api_basic_authorize
+RSpec.describe "Regions API", :regions do
+  context "authorization", :authorization do
+    it "forbids access to regions without an appropriate role" do
+      api_basic_authorize
 
-    get(api_regions_url)
+      get(api_regions_url)
 
-    expect(response).to have_http_status(:forbidden)
+      expect(response).to have_http_status(:forbidden)
+    end
+
+    it "forbids access to a region resource without an appropriate role" do
+      api_basic_authorize
+
+      region = FactoryBot.create(:miq_region, :region => "2")
+
+      get(api_region_url(nil, region))
+
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
+
+  context "get", :get do
+    it "allows GETs of a region" do
+      api_basic_authorize action_identifier(:regions, :read, :resource_actions, :get)
+
+      region = FactoryBot.create(:miq_region, :region => "2")
+
+      get(api_region_url(nil, region))
+
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(
+        "href" => api_region_url(nil, region),
+        "id"   => region.id.to_s
+      )
+    end
   end
 
-  it "forbids access to a region resource without an appropriate role" do
-    api_basic_authorize
+  context "edit", :edit do
+    it "can update a region with POST" do
+      api_basic_authorize action_identifier(:regions, :edit)
+
+      region = FactoryBot.create(:miq_region, :description => "Current Region description")
+
+      post api_region_url(nil, region), :params => gen_request(:edit, :description => "New Region description")
+
+      expect(response).to have_http_status(:ok)
+      region.reload
+      expect(region.description).to eq("New Region description")
+    end
+
+    it "will fail if you try to edit forbidden fields" do
+      api_basic_authorize action_identifier(:regions, :edit)
 
-    region = FactoryBot.create(:miq_region, :region => "2")
+      region = FactoryBot.create(:miq_region, :description => "Current Region description")
 
-    get(api_region_url(nil, region))
+      post api_region_url(nil, region), :params => gen_request(:edit, :created_at => Time.now)
 
-    expect(response).to have_http_status(:forbidden)
+      expect(response).to have_http_status(:bad_request)
+    end
   end
 
-  it "allows GETs of a region" do
-    api_basic_authorize action_identifier(:regions, :read, :resource_actions, :get)
+  context "delete", :delete do
+    it "can delete a region with POST" do
+      api_basic_authorize action_identifier(:regions, :delete)
+      region = FactoryBot.create(:miq_region)
 
-    region = FactoryBot.create(:miq_region, :region => "2")
+      expect { post api_region_url(nil, region), :params => gen_request(:delete) }.to change(MiqRegion, :count).by(-1)
+      expect(response).to have_http_status(:ok)
+    end
```

Comment:
maybe add a test for bulk deletes, POST the delete action to the collection and delete 2 regions.

Reply:
Alright, I can add that.

Reply:
Done!